src/main/entity/User.java

```diff
@@ -1,50 +1,91 @@
 package entity;
 
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * Implementation of User.
  */
-public class User {
-    private String name;
-    private String uofTEmail;
-    private String passwordHash;
-    private String username;
-
-    public User(String name, String email, String password, String username) {
-        this.name = name;
-        this.uofTEmail = email;
-        this.passwordHash = password;
-        this.username = username;
-    }
-
-    public String getName() {
-        return name;
-    }
+class User extends Location {
```

Comment:
All other entities look correct. User is a seperate class and doesn't extend abstract class Location. Its purpose is to have a local list of favorite rooms.